### ISSUE TITLE
Terraform 0.11.3 taint command can't find any resource

### ISSUE THREAD

**Issue by samuelbond** [hashicorp/terraform#17512]
### Terraform Version

```
Terraform v0.11.3
+ provider.aws v1.10.0
+ provider.external v1.0.0
```

### Terraform Configuration Files

```hcl
resource "aws_api_gateway_base_path_mapping" "custom_domain_mapping" {
  depends_on = ["aws_api_gateway_deployment.deployment"]
  api_id      = "${aws_api_gateway_rest_api.rest_api.id}"
  stage_name  = "${aws_api_gateway_deployment.deployment.stage_name}"
  domain_name = "${aws_route53_record.custom_domain_alias_record.name}"
  base_path   = "${var.stage_name}"
}
```

### Expected Behavior

When I run `terraform taint aws_api_gateway_base_path_mapping.custom_domain_mapping` the usual output is `The resource aws_api_gateway_base_path_mapping.custom_domain_mapping in the module root has been marked as tainted!`
### Actual Behavior

After upgrading to `version 0.11.3`, same command and exactly same resource produces
`The resource aws_api_gateway_base_path_mapping.custom_domain_mapping couldn't be found in the module root.`
### Steps to Reproduce

`terraform init`
`terraform taint aws_api_gateway_base_path_mapping.custom_domain_mapping`

**Comment by jbardin** [comment]
Hi @samuelbond,

Sorry you're having trouble with this, but I'm not able to reproduce it myself. 

Does `terraform plan` show any pending changes to your resources?
Does `terraform list aws_api_gateway_base_path_mapping.custom_domain_mapping` show any output at all?

Can you locate the resource in the state file, and verify the name and what the `"path"` entry is in the json object?

**Comment by samuelbond** [comment]
Hi @jbardin 
Thanks for your response, did you mean `terraform state list aws_api_gateway_base_path_mapping.custom_domain_mapping` instead ?

**Comment by samuelbond** [comment]
@jbardin I found out why it failed to taint. The previous resource was destroyed and before terraform could recreate it an error occurred so that resource didn't exist anymore and when i ran taint terraform just couldn't find it, meaning terraform will only taint an existing resource.

I have resolved this by deploying my infrastructure again successfully and then when i run taint, terraform is able to taint the resource now because it already exist.
